keanu-project/src/main/java/io/improbable/keanu/model/SamplingModelFitter.java

```diff
@@ -0,0 +1,52 @@
+package io.improbable.keanu.model;
+
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.algorithms.PosteriorSamplingAlgorithm;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.Vertex;
+
+public class SamplingModelFitter<INPUT, OUTPUT> implements ModelFitter<INPUT, OUTPUT> {
+
+    private final ModelGraph<INPUT, OUTPUT> modelGraph;
+    private final PosteriorSamplingAlgorithm samplingAlgorithm;
+    private final int sampleCount;
+    private final int dropCount;
+
+    /**
+     * This fitter uses a {@link PosteriorSamplingAlgorithm}, in contrast to the {@link MAPModelFitter} and {@link MaximumLikelihoodModelFitter}, which use gradient methods.
+     *
+     * The model's latent vertices will have their values set to the average over the samples.
+     * A down-sample of 2 is used (not configurable).
```

Comment:
Just curious - why is is we use a down-sampling rate of 2?

Reply:
Down-sampling is to avoid cross-correlation in the samples. I simply chose 2 because it's what some of the other tests do. 

We could make this configurable but I didn't want to clutter the API. Any thoughts @gordoncaleb ? I'm also planning to discuss it with @johannespetrat when he's got some time next week.

Reply:
This comment is now obsolete - I will remove it. I'm not doing any downsampling inside the fitter any more.

Reply:
this is obsolete now - don't do the down-sampling as part of the fitting any more